openwisp_monitoring/device/api/views.py

```diff
@@ -165,6 +167,12 @@ def post(self, request, pk):
         except ValidationError as e:
             logger.info(e.message)
             return Response(e.message, status=status.HTTP_400_BAD_REQUEST)
+        time_obj = request.query_params.get('time')
+        if time_obj:
+            time = datetime.strptime(time_obj, '%d-%m-%Y_%H:%M:%S.%f')
+            self.time = time.replace(tzinfo=UTC)
+        else:
+            self.time = None
```

Comment:
I think we should pass this around as argument instead of using an attribute, it's an argument specified by the sender, it's really weird to set it as an attribute.